cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -457,6 +494,23 @@ func (m *McmManager) DeleteMachines(machines []*Ref) error {
 		}
 	}
 
+	var md *v1alpha1.MachineDeployment
+	retryDeadline := time.Now().Add(maxRetryDeadline)
+	for {
```

Comment:
why do you now have 2 for loops whereas in the original code you only had one?

Reply:
in the original code it was like this
--> loop1  -> Update the prio of machines to 1
---> loop2 -> Now Fetch the machineDeployment object and update the replicas

with current requirements where we need to confirm that the machineDeployment is rolling updating or not we need the md object even before the machine prio updates . So 

---> loop1 -> Fetch the md object and check if rolling update happening
---> loop2 -> if not , then update machine prios to 1
---> loop3 -> update the machineDeployment replicas